### ISSUE TITLE
Enhancement: [no-floating-promises] need a configuration that would allow to exclude an object from this rule

### ISSUE THREAD

**Issue by budarin** [typescript-eslint/typescript-eslint#5844]
### Before You File a Proposal Please Confirm You Have Done The Following...

- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22enhancement%3A+plugin+rule+option%22) and found none that match my proposal.
- [X] I have searched the [current rule list](https://typescript-eslint.io/rules/#supported-rules) and found no rules that match my proposal.
- [X] I have [read the FAQ](https://typescript-eslint.io/docs/linting/troubleshooting) and my problem is not listed.

### My proposal is suitable for this project

- [X] I believe my proposal would be useful to the broader TypeScript community (meaning it is not a niche proposal).

### Link to the rule's documentation

https://typescript-eslint.io/rules/no-floating-promises/

### Description

[Fastify.reply](https://github.com/fastify/fastify/blob/68914519a3fd65ce65ba8db8962940eecf7f0fc8/types/reply.d.ts#L63) object has `then` method, but reply is not a Promise and its methods are not asynchronous

typescript-eslint marks each method of the `reply` object as asynchronous and requires it to be preceded by an `await` call

### Fail

```typescript
import { FastifyReply, FastifyRequest } from 'fastify';

function setNotFound(req: FastifyRequest, res: FastifyReply, done: HookHandlerDoneFunction): void {
    res.status(404);
    res.send('Not found');

    done();
}
```
![image](https://user-images.githubusercontent.com/8055157/196480832-ff77f946-5f43-4b26-8926-731f795a4dd9.png)

### Pass

```typescript
import { FastifyReply, FastifyRequest } from 'fastify';

function setNotFound(req: FastifyRequest, res: FastifyReply, done: HookHandlerDoneFunction): void {
    void res.status(404);
    void res.send('Not found');

    done();
}
```

### Additional Info
The plugin should look at what the object's method returns, and not just whether the object has a `then` method.

Or it is necessary to configure the rule to be able to exclude this object/methods from consideration by this rule

**Comment by bradzacher** [comment]
duplicate of #2640
